src/main/webapp/script.js

```diff
@@ -12,17 +12,34 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+// Create the script tag, set the appropriate attributes
+var script = document.createElement('script');
+script.src = 'https://maps.googleapis.com/maps/api/js?key=AIzaSyDZbsCASX6bRFXIojx2BUzeBcwI12oeuyU&callback=initMap';
+script.async = true;
+
+var map;
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+// Attach your callback function to the `window` object
+window.initMap = function() {
+  map = new google.maps.Map(document.getElementById("maps"), {
+      center: { lat: 40.7484405, lng: -73.9878584 },
+      zoom: 4
+  });
+};
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
-}
+// Append the 'script' element to 'head'
+document.head.appendChild(script);
+
+/**
+ * The onload function sets the height for the
+ * maps DIV element.
+ */
+window.onload = function() {
+    const headerHeight = document.getElementById("header").offsetHeight;
+    const footerHeight = document.getElementById("footer").offsetHeight;
+    var mapsHeight = document.body.scrollHeight - headerHeight - footerHeight;
```

Comment:
let*

Also no need to do this in JS. Use css classes

Reply:
I've changed it to let. I need to use JS instead of CSS because I can't obtain the height of other elements in CSS. The height of the header and footer change depending on what's in it and the kind of device being used that's why I used JS to determine the height of the map when a user opens the page instead of presetting it.

Reply:
Specify the height in px. 
Set width to 100%

Reply:
I didn't get the message you were trying to convey until I found out that CSS had a calc function that allows arithmetic operations between different CSS units (very cool).